activerecord/lib/active_record/reflection.rb

```diff
@@ -154,6 +154,10 @@ def build_association(attributes, &block)
         klass.new(attributes, &block)
       end
 
+      def scope_for(relation, owner = nil)
+        relation.instance_exec(owner, &scope) || relation
+      end
```

Comment:
`AbstractReflection ` is shared to all reflections (includes `AggregateReflection`).
How about adding `delegate :scope_for` in `PolymorphicReflection` instead, to affect to the reflection only?